rpc/block.go

```diff
@@ -145,6 +148,18 @@ type BlockWithReceipts struct {
 //
 // It follows the specification defined here:
 // https://github.com/starkware-libs/starknet-specs/blob/a789ccc3432c57777beceaa53a34a7ae2f25fda0/api/starknet_api_openrpc.json#L500
+
+type BlockWithTxHashesandReceipts struct {
+	Status BlockStatus `json:"status,omitempty"`
+	BlockHeader
+	TxnHashes []*felt.Felt `json:"transactions"`
+	Transactions []TransactionWithReceipt `json:"transactions"`
```

Comment:
Note: you have two fields with the same json-tag which will cause marshalling errors